go/libraries/doltcore/sqle/dtables/column_diff_table.go

```diff
@@ -0,0 +1,649 @@
+func (itr *doltColDiffCommitHistoryRowItr) calculateTableChanges(ctx context.Context, commit *doltdb.Commit) ([]tableColChange, error) {
+	if len(commit.DatasParents()) == 0 {
+		return nil, nil
+	}
+
+	toRootValue, err := commit.GetRootValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parent, err := itr.ddb.ResolveParent(ctx, commit, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	fromRootValue, err := parent.GetRootValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	deltas, err := diff.GetTableDeltas(ctx, fromRootValue, toRootValue)
+	if err != nil {
+		return nil, err
+	}
+
+	tableChanges := make([]tableColChange, len(deltas))
+	for i := 0; i < len(deltas); i++ {
+		change, err := processTableColDelta(itr.ctx, itr.ddb, deltas[i])
+		if err != nil {
+			return nil, err
+		}
+
+		tableChanges[i] = *change
+	}
+
+	// Not all commits mutate tables (e.g. empty commits)
+	if len(tableChanges) == 0 {
+		return nil, nil
+	}
+
+	return tableChanges, nil
+}
+
+// Close closes the iterator.
+func (itr *doltColDiffCommitHistoryRowItr) Close(*sql.Context) error {
+	return nil
+}
+
+// tableColChange is an internal data structure used to hold the results of processing
+// a diff.TableDelta structure into the output data for this system table.
+type tableColChange struct {
+	tableName    string
+	colNames     []string
+	dataChange   bool
+	schemaChange bool
+}
+
+// processTableColDelta processes the specified TableDelta to determine what kind of change it was (i.e. table drop,
+// table rename, table create, or data update) and returns a tableChange struct representing the change.
+func processTableColDelta(ctx *sql.Context, ddb *doltdb.DoltDB, delta diff.TableDelta) (*tableColChange, error) {
+	// Dropping a table is always a schema change, and also a data change if the table contained data
+	if delta.IsDrop() {
+		isEmpty, err := isTableDataEmpty(ctx, delta.FromTable)
+		if err != nil {
+			return nil, err
+		}
+
+		return &tableColChange{
+			tableName:    delta.FromName,
+			colNames:     delta.FromSch.GetAllCols().GetColumnNames(),
+			dataChange:   !isEmpty,
+			schemaChange: true,
+		}, nil
+	}
+
+	// Renaming a table is always a schema change, and also a data change if the table data differs
+	if delta.IsRename() {
+		dataChanged, err := delta.HasHashChanged()
+		if err != nil {
+			return nil, err
+		}
+
+		return &tableColChange{
+			tableName:    delta.ToName,
+			colNames:     delta.ToSch.GetAllCols().GetColumnNames(),
+			dataChange:   dataChanged,
+			schemaChange: true,
+		}, nil
+	}
+
+	// Creating a table is always a schema change, and also a data change if data was inserted
+	if delta.IsAdd() {
+		isEmpty, err := isTableDataEmpty(ctx, delta.ToTable)
+		if err != nil {
+			return nil, err
+		}
+
+		return &tableColChange{
+			tableName:    delta.ToName,
+			colNames:     delta.ToSch.GetAllCols().GetColumnNames(),
+			dataChange:   !isEmpty,
+			schemaChange: true,
+		}, nil
+	}
+
+	dataChanged, err := delta.HasHashChanged()
```

Comment:
For this one, it would be good to use `delta.HasDataChanged()`. Taylor found a correctness bug here the other day... the hash for the table includes the schema, so if only the schema changed, then this should also show as true. The `HasDataChanged()` function she added should give you the correct result here.